### PR TITLE
[config] Add various tests to improve code coverage in config

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR helps to improve the code coverage of tests in the configuration files by adding several simple tests.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

'cargo xtest' seems to be happy with the new tests.

## Related PRs

None.

### CODE REVIEW

File: config/src/config/mod.rs [hunk 1]
```diff
@@ -118,7 +119,7 @@ impl RoleType {
     }
 }
 
-impl std::str::FromStr for RoleType {
+impl FromStr for RoleType {
```
Comment (davidiw) [inline config/src/config/mod.rs]: Did this improve coverage? or did we need the trait for other purposes?

**Comment by davidiw** [comment]
@bors-libra r+

shockingly the test config test can actually be valuable if we become overly dependent on poorly defined behaviors of that config... part of my wonderful experience in refactoring these files :/

File: config/src/config/mod.rs (hunk 1 above)
Comment (JoshLind) [reply config/src/config/mod.rs]: @davidiw, we needed the trait for tests I added for the role type conversion.

**Comment by JoshLind** [comment]
Thanks for submitting this, David!

**Pull request closed without merging**